src/drivers/uavcannode/Publishers/GlobalNavigationSolution.hpp

```diff
@@ -0,0 +1,113 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#pragma once
+
+#include <cmath>
+
+#include "UavcanPublisherBase.hpp"
+
+#include <uavcan/navigation/GlobalNavigationSolution.hpp>
+
+#include <uORB/SubscriptionCallback.hpp>
+#include <uORB/topics/vehicle_global_position.h>
+
+namespace uavcannode
+{
+
+class GlobalNavigationSolution :
+	public UavcanPublisherBase,
+	public uORB::SubscriptionCallbackWorkItem,
+	private uavcan::Publisher<uavcan::navigation::GlobalNavigationSolution>
+{
+public:
+	GlobalNavigationSolution(px4::WorkItem *work_item, uavcan::INode &node) :
+		UavcanPublisherBase(uavcan::navigation::GlobalNavigationSolution::DefaultDataTypeID),
+		uORB::SubscriptionCallbackWorkItem(work_item, ORB_ID(vehicle_global_position)),
+		uavcan::Publisher<uavcan::navigation::GlobalNavigationSolution>(node)
+	{
+		this->setPriority(uavcan::TransferPriority::OneLowerThanHighest);
+	}
+
+	void PrintInfo() override
+	{
+		if (uORB::SubscriptionCallbackWorkItem::advertised()) {
+			printf("\t%s -> %s:%d\n",
+			       uORB::SubscriptionCallbackWorkItem::get_topic()->o_name,
+			       uavcan::navigation::GlobalNavigationSolution::getDataTypeFullName(),
+			       id());
+		}
+	}
+
+	void BroadcastAnyUpdates() override
+	{
+		using uavcan::navigation::GlobalNavigationSolution;
+
+		// vehicle_global_position -> uavcan::navigation::GlobalNavigationSolution
```

Comment:
It would be better to do this with vehicle_local_position. The vehicle_global_position is just the local NED estimate reprojected with the ref_lat, ref_lon fields available in vehicle_local_position.

https://github.com/PX4/PX4-Autopilot/blob/6eae9fb37107ba55d52ab7e3189b1c403cea45a6/msg/VehicleLocalPosition.msg#L47-L53

Reply:
Updated to vehicle_local_position and force pushed.